ykrt/src/mt.rs

```diff
@@ -379,8 +379,11 @@ impl MTThread {
                             let ptr: *mut CompiledTrace<I> = Box::into_raw(Box::new(ct));
                             let new_state = ptr as usize | PHASE_COMPILED;
                             loc.state.store(new_state, Ordering::Release);
-                            // Go around the loop so that the PHASE_COMPILED case below can extract
-                            // the function.
+
+                            let bct = unsafe { Box::from_raw(ptr) };
```

Comment:
Would this be better expressed as an inner function (defined above the loop) with two callsites?

Reply:
I'm not sure it makes much of a difference either way to be honest. We'd save 4 lines, but add 6 lines...

Reply:
I ask not because I want to save lines, but because then we then have only one place to make concurrency mistakes. I can imagine updating the code in one place, and not in the other -- that's totally the kind of mistake I'd make.

Reply:
I have tried breaking some of the code (including `type`s out), but it's tended to slightly reduce (or, at least, not improve) readability. I will probably try again later, but perhaps after we've gone through some more PRs, when commonalities in code might become more apparent.